File: python-redis-demo/examples/example04_hashes.py
import redis

cache = redis.StrictRedis(host='192.168.1.125', port=6379, db=0)

cache.hset("user#1", "name", "Peter")
cache.hset("user#1", "job", "politician")

name = cache.hget("user#1", "name")

fields = cache.hgetall("user#1")
print(type(fields))
print(fields)

job = fields.get("job".encode())
print("job:", job.decode())

for i in fields:
    print(i.decode(), fields.get(i).decode())

File: redis-jedis-demo/src/main/java/com/akatkar/java/learning/redis/demo/Example04_Hashes.java
package com.akatkar.java.learning.redis.demo;

import java.util.Map;
import redis.clients.jedis.Jedis;

import javax.xml.bind.SchemaOutputResolver;

import static com.akatkar.java.learning.redis.demo.Config.*;

/**
 * Redis Hashes are mapping between String fields and String values:
 */
public class Example04_Hashes {

    public static void main(String[] args) {

        try(Jedis jedis = new Jedis(REDIS_HOST)) {
            jedis.hset("user#1", "name", "Peter");
            jedis.hset("user#1", "job", "politician");

            // retrieve name of user1 from redis
            String name = jedis.hget("user#1", "name");
            System.out.println(name);
            // retrieve all fields of user1 from redis
            Map<String, String> fields = jedis.hgetAll("user#1");
            String job = fields.get("job");
            System.out.println(job);
        }
    }
}
